process/transaction/shardProcess.go

```diff
@@ -173,6 +173,13 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 			}
 		}
 
+		if errors.Is(err, process.ErrUserNameDoesNotMatch) && txProc.flagRelayedTx.IsSet() {
+			receiptErr := txProc.executingFailedTransaction(tx, acntSnd, err)
+			if receiptErr != nil {
+				return vmcommon.UserError, receiptErr
```

Comment:
For ErrInsufficientFunds error above, if executingFailedTransaction fails it is returned (0, receiptErr) and here (vmcommon.UserError, receiptErr), it is ok?

Reply:
CC: @sasurobert

Reply:
Yes it is. User code return is not used on the return, only in tests.